firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/ImpressionStorageClient.java

```diff
@@ -96,4 +98,39 @@ private void clearInMemCache() {
         .map(CampaignImpression::getCampaignId)
         .contains(campaignId);
   }
+
+  /**
+   * Clears impressions for all campaigns found in the provided {@link
+   * FetchEligibleCampaignsResponse}
+   */
+  public Completable clearImpressions(FetchEligibleCampaignsResponse response) {
+    HashSet<String> idsToClear = new HashSet<>();
+    for (CampaignProto.ThickContent content : response.getMessagesList()) {
+      String id =
+          content.getPayloadCase().equals(CampaignProto.ThickContent.PayloadCase.VANILLA_PAYLOAD)
+              ? content.getVanillaPayload().getCampaignId()
+              : content.getExperimentalPayload().getCampaignId();
+      idsToClear.add(id);
+    }
+    Logging.logd("Potential impressions to clear: " + idsToClear.toString());
```

Comment:
Did we test if this `HashSet.toString` logs the items and not the reference to the array?

Reply:
yeah it logs the items. it logs a reference to the array if the array is empty but if it contains items it logs the items.